File: Lab-7/examples/nextjs-scaffold/lib/guardrails.ts
import type { AgentState } from "./agent-state";

export function requiresHumanApproval(state: AgentState): boolean {
  const text = state.userRequest.toLowerCase();

  return [
    "send",
    "email",
    "submit",
    "delete",
    "book",
    "purchase",
  ].some(term => text.includes(term));
}

export function buildCheckpointMessage(state: AgentState): string {
  return `Approval required before action runs: ${state.userRequest}`;
}

File: Lab-7/examples/fastapi-scaffold/services/guardrails.py
from schemas.agent_state import AgentState


def requires_human_approval(state: AgentState) -> bool:
    request = state.get("user_request", "").lower()

    risky_terms = [
        "send",
        "email",
        "submit",
        "delete",
        "book",
        "purchase",
    ]

    return any(term in request for term in risky_terms)


def build_checkpoint_message(state: AgentState) -> str:
    return (
        "Approval required before this action runs. "
        f"Current request: {state.get('user_request', '')}"
    )
